app/js/reducers/fetchOrdersReducer.js

```diff
@@ -68,6 +68,14 @@ const fetchOrdersReducer = (state = initialState, action) => {
       };
     }
 
+    case 'DISCONTINUE_ACTIVE_DRUG_ORDER': {
+      return {
+        ...state,
+        filteredOrders: state.filteredOrders
+          .filter(order => order.orderNumber !== action.orderNumber),
+      };
```

Comment:
@Efosaok when you mean update the original results array, I don't understand cos you're not being clear enough. Would you mind clarifying that ?

Reply:
In this reducer, the state is something like this,

```
{
   orders: {
      results: [1, 2, 3],
      links: 'some-random-link',
      totalCount: 23,
   },
   filteredOrders: [1, 2, 3],
   status: {
     added: true,
     error: false,
  },
}
```

`orders.results` is the original array I am referring to here, the `filteredOrders` array is usually manipulated based on the filter query but the `orders.results` array always contains the initial data from the database. so while you update the `filteredOrders` array due to a discard, you would have to update that array as well, or else it could lead to an undesired behaviour when sorting and filtering after discarding an orderable.

Reply:
@Efosaok I saw that, so filteredOrders would hold the already sorted and filtered values in state right ? so if I filter a value from the sorted filteredValues, I don't think it would cause any undesirable behaviour because it's already being sorted for one so filtering one out doesn't change anything apart from their positions which is sorted already. so that don't change anything, but I would like you to run the branch, break stuff and test out, just to see what undesirable behaviours it could cause or mention any if you have any thoughts on it at the moment.